retry/lib/retry.dart

```diff
@@ -179,10 +196,27 @@ Future<T> retry<T>(
   int maxAttempts = 8,
   FutureOr<bool> Function(Exception)? retryIf,
   FutureOr<void> Function(Exception)? onRetry,
+  FutureOr<T> Function(Exception)? orElse,
 }) =>
     RetryOptions(
       delayFactor: delayFactor,
       randomizationFactor: randomizationFactor,
       maxDelay: maxDelay,
       maxAttempts: maxAttempts,
-    ).retry(fn, retryIf: retryIf, onRetry: onRetry);
+    ).retry(fn, retryIf: retryIf, onRetry: onRetry, orElse: orElse);
+
+/// throw [RetryExhaustedException] when retries is exhausted.
```

Comment:
> This is a breaking change, even if `orElse` is omitted.
> 
> Is it really better to get a retries exhausted exception, rather than throwing the last exception that was caught?
> 
> We'd need to investigate of breaking this package would affect many users.

I think the reason you need `RetryExhaustException` is that when you need to do subsequent processing for a retry failure, you need to repeatedly use the logic in `retryIf` to determine if the exception is a failed retry.

Maybe not needing `RetryExhaustException`, `orElse` to handle only the `retryIf` exceptions is a better way?

But here the `RetryExhaustException` throwing should rely on `retryIf`, a little embarrassing😅.